From a546c94e9669ef928e5a13e9d953623838596e8c Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Fri, 19 Mar 2021 12:11:50 +0530
Subject: [PATCH 1327/1921] Revert "octeontx2: Re-enable FLR and MBOX
 interrupts"

This reverts commit 7d1dc17c5092805acfa1fbf214544cc04f7bb8a7.
PF_RST on a FLR is causing RVU_PF_VF_BAR4_ADDR also getting cleared.
Due to that RVU_PF_VF_BAR4_ADDR is also getting cleared which would
break PF::VF communication

Change-Id: Ia263de779f135f89619f35982691c26d904f7468
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/48087
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/soc/marvell/octeontx2-npa/npa.c    | 9 ---------
 drivers/soc/marvell/octeontx2-rm/otx2_rm.c | 8 +-------
 drivers/soc/marvell/octeontx2-sdp/sdp.c    | 9 ---------
 3 files changed, 1 insertion(+), 25 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2-npa/npa.c b/drivers/soc/marvell/octeontx2-npa/npa.c
index f996cf0190b2..08e0c186a653 100644
--- a/drivers/soc/marvell/octeontx2-npa/npa.c
+++ b/drivers/soc/marvell/octeontx2-npa/npa.c
@@ -466,15 +466,6 @@ static void npa_pfvf_flr_handler(struct work_struct *work)
 	otx2_enable_afpf_mbox_intr(npa);
 	npa_write64(npa, BLKADDR_RVUM, 0, RVU_PF_VFTRPENDX(vf->vf_id / 64),
 		    BIT_ULL(vf->intr_idx));
-
-	/* Re-enable MBOX and FLR interrupt as it gets cleared
-	 * in HWVF_RST reset
-	 */
-	npa_write64(npa, BLKADDR_RVUM, 0, RVU_PF_VFFLR_INT_ENA_W1SX(vf->vf_id),
-		   BIT_ULL(vf->intr_idx));
-	npa_write64(npa, BLKADDR_RVUM, 0,
-		   RVU_PF_VFPF_MBOX_INT_ENA_W1SX(vf->vf_id / 64),
-		   BIT_ULL(vf->intr_idx));
 }
 
 static void npa_pfvf_mbox_handler_up(struct work_struct *work)
diff --git a/drivers/soc/marvell/octeontx2-rm/otx2_rm.c b/drivers/soc/marvell/octeontx2-rm/otx2_rm.c
index 8f8067cf4757..ea76bd3db6e4 100644
--- a/drivers/soc/marvell/octeontx2-rm/otx2_rm.c
+++ b/drivers/soc/marvell/octeontx2-rm/otx2_rm.c
@@ -458,15 +458,9 @@ static void rm_pfvf_flr_handler(struct work_struct *work)
 	rm_write64(rm, BLKADDR_RVUM, 0, RVU_PF_VFTRPENDX(idx),
 		   BIT_ULL(vf->intr_idx));
 
-	/* Re-enable MBOX, FLR, and ME interrupt as it gets cleared
-	 * in HWVF_RST reset
-	 */
-	rm_write64(rm, BLKADDR_RVUM, 0, RVU_PF_VFFLR_INT_ENA_W1SX(idx),
-		   BIT_ULL(vf->intr_idx));
+	/* Re-enable ME interrupt as it gets cleared in HWVF_RST reset */
 	rm_write64(rm, BLKADDR_RVUM, 0, RVU_PF_VFME_INT_ENA_W1SX(idx),
 		   BIT_ULL(vf->intr_idx));
-	rm_write64(rm, BLKADDR_RVUM, 0, RVU_PF_VFPF_MBOX_INT_ENA_W1SX(idx),
-		   BIT_ULL(vf->intr_idx));
 }
 
 static void rm_pfvf_mbox_handler_up(struct work_struct *work)
diff --git a/drivers/soc/marvell/octeontx2-sdp/sdp.c b/drivers/soc/marvell/octeontx2-sdp/sdp.c
index 699fcac2404c..152da151a960 100644
--- a/drivers/soc/marvell/octeontx2-sdp/sdp.c
+++ b/drivers/soc/marvell/octeontx2-sdp/sdp.c
@@ -471,15 +471,6 @@ static void sdp_pfvf_flr_handler(struct work_struct *work)
 	enable_af_mbox_int(sdp->pdev);
 	sdp_write64(sdp, BLKADDR_RVUM, 0, RVU_PF_VFTRPENDX(vf->vf_id / 64),
 		   BIT_ULL(vf->intr_idx));
-
-	/* Re-enable MBOX and FLR interrupt as it gets cleared
-	 * in HWVF_RST reset
-	 */
-	sdp_write64(sdp, BLKADDR_RVUM, 0, RVU_PF_VFFLR_INT_ENA_W1SX(vf->vf_id),
-		   BIT_ULL(vf->intr_idx));
-	sdp_write64(sdp, BLKADDR_RVUM, 0,
-		   RVU_PF_VFPF_MBOX_INT_ENA_W1SX(vf->vf_id / 64),
-		   BIT_ULL(vf->intr_idx));
 }
 
 static void sdp_pfvf_mbox_handler_up(struct work_struct *work)
-- 
2.31.1

